MaaS 3scale example with minio - developer portal does not come up

### System Info

OCP 4.19
RHOAI 2.19

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### 🐛 Describe the bug

After modifying storageClassName to "" for enabling maas with minio as storage solution, I ran the bootstrap.sh script and waited for the maas components to sync in argocd.

Afterwards, I was able to successfully go to the 3scale admin portal. I copied the developer portal access code then went to the developer portal with the access code but I got a HTTP 404 with message:

`This maas.apps.cluster-xsjml.xsjml.sandbox5442.opentlc.com page can’t be found`

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/ab186d96-d313-41fd-b170-47173737c3ee" />

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/b6bbad43-260b-4ba1-9a05-60d7a163309d" />

### Error logs

The issue could be with the upload-cms pipeline. The runs are failing as seen below:

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/ee6cd12b-0b0b-4db1-a4a1-19020d19ac89" />

The first run passed on the `clone-contents` task but fails on the `upload-cms` task with error `responseBody='{"errors":{"system_name":["reserved for internal use"]}}`
Full logs are below:

```
Uploading SECTION /...
Uploading SECTION /applications/...
Uploading SECTION /services/...
Uploading SECTION /login/...
Uploading SECTION /messages/inbox/...
Uploading SECTION /user/...
Uploading SECTION /accounts/...
Uploading SECTION /javascripts/...
Uploading SECTION /dashboards/...
Uploading SECTION /errors/...
Uploading SECTION /applications/alerts/...
Uploading SECTION /stats/...
Uploading SECTION /invitations/...
Uploading SECTION /messages/trash/...
Uploading SECTION /accounts/payment_gateways/...
Uploading SECTION /password/...
Uploading SECTION /accounts/invitee_signups/...
Uploading SECTION /shared/...
Uploading SECTION /signup/...
Uploading SECTION /account_plans/...
Uploading SECTION /search/...
Uploading SECTION /images/...
Uploading SECTION /messages/...
Uploading SECTION /css/...
Uploading SECTION /messages/outbox/...
Uploading SECTION /accounts/users/...
Uploading SECTION /account/...
Uploading SECTION /invoices/...
Uploading TEMPLATE /l_error.html.liquid...
Uploading TEMPLATE /accounts/users/index.html.liquid...
Uploading TEMPLATE /messages/_menu.html.liquid...
com.fwmotion.threescale.cms.exception.ThreescaleCmsApiException
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.handleApiException(ThreescaleCmsClientImpl.java:113)
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.handleApiErrors(ThreescaleCmsClientImpl.java:75)
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.handleApiErrors(ThreescaleCmsClientImpl.java:80)
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.saveNewTemplate(ThreescaleCmsClientImpl.java:365)
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.savePartial(ThreescaleCmsClientImpl.java:341)
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.save(ThreescaleCmsClientImpl.java:286)
	at com.fwmotion.threescale.cms.cli.UploadCommand.performUpload(UploadCommand.java:239)
	at com.fwmotion.threescale.cms.cli.UploadCommand.call(UploadCommand.java:199)
	at com.fwmotion.threescale.cms.cli.UploadCommand.call(UploadCommand.java:19)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at io.quarkus.picocli.runtime.PicocliRunner$EventExecutionStrategy.execute(PicocliRunner.java:26)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at io.quarkus.picocli.runtime.PicocliRunner.run(PicocliRunner.java:40)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:132)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:61)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
Caused by: ApiException{code=422, responseHeaders={date=[Thu, 25 Sep 2025 05:27:03 GMT], x-request-id=[ef06520f-dbab-47ba-ad1f-b49917534f92], transfer-encoding=[chunked], vary=[Accept-Encoding], x-frame-options=[DENY], x-served-by=[system-app-765f57c9ff-wfvlp], strict-transport-security=[max-age=31536000; includeSubDomains], set-cookie=[052650289121bf47ef52d4579d835bb5=07af743efa2caff3918038b294b7952d; path=/; HttpOnly; Secure; SameSite=None], x-content-type-options=[nosniff], x-xss-protection=[1; mode=block], x-runtime=[0.023242], content-type=[application/json; charset=utf-8], cache-control=[no-cache]}, responseBody='{"errors":{"system_name":["reserved for internal use"]}}'}
	at com.redhat.threescale.rest.cms.ApiClient.processResponse(ApiClient.java:913)
	at com.redhat.threescale.rest.cms.ApiClient.invokeAPI(ApiClient.java:1000)
	at com.redhat.threescale.rest.cms.api.TemplatesApi.createTemplate(TemplatesApi.java:157)
	at com.redhat.threescale.rest.cms.api.TemplatesApi.createTemplate(TemplatesApi.java:76)
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.lambda$saveNewTemplate$9(ThreescaleCmsClientImpl.java:365)
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.handleApiErrors(ThreescaleCmsClientImpl.java:73)
	... 27 more
```

I attempted a manual rerun but got near results with the second failing with message `responseBody='{"errors":{"system_name":["has already been taken"],"path":["has already been taken"]}}'` 
Full logs are below: 

```
Uploading SECTION /...
Uploading SECTION /applications/...
Uploading SECTION /services/...
Uploading SECTION /login/...
Uploading SECTION /messages/inbox/...
Uploading SECTION /user/...
Uploading SECTION /accounts/...
Uploading SECTION /javascripts/...
Uploading SECTION /dashboards/...
Uploading SECTION /errors/...
Uploading SECTION /applications/alerts/...
Uploading SECTION /stats/...
Uploading SECTION /invitations/...
Uploading SECTION /messages/trash/...
Uploading SECTION /accounts/payment_gateways/...
Uploading SECTION /password/...
Uploading SECTION /accounts/invitee_signups/...
Uploading SECTION /shared/...
Uploading SECTION /signup/...
Uploading SECTION /account_plans/...
Uploading SECTION /search/...
Uploading SECTION /images/...
Uploading SECTION /messages/...
Uploading SECTION /css/...
Uploading SECTION /messages/outbox/...
Uploading SECTION /accounts/users/...
Uploading SECTION /account/...
Uploading SECTION /invoices/...
Uploading TEMPLATE /l_error.html.liquid...
Uploading TEMPLATE /accounts/users/index.html.liquid...
com.fwmotion.threescale.cms.exception.ThreescaleCmsApiException
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.handleApiException(ThreescaleCmsClientImpl.java:113)
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.handleApiErrors(ThreescaleCmsClientImpl.java:75)
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.handleApiErrors(ThreescaleCmsClientImpl.java:80)
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.saveNewTemplate(ThreescaleCmsClientImpl.java:365)
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.savePage(ThreescaleCmsClientImpl.java:328)
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.save(ThreescaleCmsClientImpl.java:284)
	at com.fwmotion.threescale.cms.cli.UploadCommand.performUpload(UploadCommand.java:239)
	at com.fwmotion.threescale.cms.cli.UploadCommand.call(UploadCommand.java:199)
	at com.fwmotion.threescale.cms.cli.UploadCommand.call(UploadCommand.java:19)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at io.quarkus.picocli.runtime.PicocliRunner$EventExecutionStrategy.execute(PicocliRunner.java:26)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at io.quarkus.picocli.runtime.PicocliRunner.run(PicocliRunner.java:40)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:132)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:61)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
Caused by: ApiException{code=422, responseHeaders={date=[Thu, 25 Sep 2025 05:41:57 GMT], x-request-id=[f72d13aa-c4aa-472e-9162-3b86701859e4], transfer-encoding=[chunked], vary=[Accept-Encoding], x-frame-options=[DENY], x-served-by=[system-app-765f57c9ff-wfvlp], strict-transport-security=[max-age=31536000; includeSubDomains], set-cookie=[052650289121bf47ef52d4579d835bb5=07af743efa2caff3918038b294b7952d; path=/; HttpOnly; Secure; SameSite=None], x-content-type-options=[nosniff], x-xss-protection=[1; mode=block], x-runtime=[0.021089], content-type=[application/json; charset=utf-8], cache-control=[no-cache]}, responseBody='{"errors":{"system_name":["has already been taken"],"path":["has already been taken"]}}'}
	at com.redhat.threescale.rest.cms.ApiClient.processResponse(ApiClient.java:913)
	at com.redhat.threescale.rest.cms.ApiClient.invokeAPI(ApiClient.java:1000)
	at com.redhat.threescale.rest.cms.api.TemplatesApi.createTemplate(TemplatesApi.java:157)
	at com.redhat.threescale.rest.cms.api.TemplatesApi.createTemplate(TemplatesApi.java:76)
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.lambda$saveNewTemplate$9(ThreescaleCmsClientImpl.java:365)
	at com.fwmotion.threescale.cms.ThreescaleCmsClientImpl.handleApiErrors(ThreescaleCmsClientImpl.java:73)
	... 27 more
```


### Expected behavior

The expected behavior is I should see this screen when opening up the developer portal:

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/e37b16f8-1e29-4f18-b441-d00f25117a5a" />

--- a/.gitignore
+++ b/.gitignore
@@ -162,3 +162,5 @@ dmypy.json
 
 # pytype static type analyzer
 .pytype/
+
+.monokle

--- a/.wordlist-txt
+++ b/.wordlist-txt
@@ -8,6 +8,7 @@ ArgoCD
 Backend
 Backends
 Backoff
+CMS
 CORS
 DataSciencePipelines
 GPUs

--- a/README.md
+++ b/README.md
@@ -92,6 +92,7 @@ helm install ai-examples ./charts/argocd-appgenerator -n openshift-gitops \
 ```bash
 # Validate Kustomize manifests and Helm charts
 ./scripts/validate_manifests.sh
+./scripts/validate_charts.sh
 
 # Install Python dependencies for linting
 pip install -r requirements.txt

--- a/examples/models-as-a-service-3scale/3scale-config/cms-upload/templates/upload-cms-eventlistener-route.yaml
+++ b/examples/models-as-a-service-3scale/3scale-config/cms-upload/templates/upload-cms-eventlistener-route.yaml
@@ -1,0 +1,11 @@
+---
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: upload-cms
+spec:
+  to:
+    kind: Service
+    name: el-upload-cms
+  tls:
+    termination: edge

--- a/examples/models-as-a-service-3scale/3scale/templates/secret-minio-s3-auth.yaml
+++ b/examples/models-as-a-service-3scale/3scale/templates/secret-minio-s3-auth.yaml
@@ -5,16 +5,16 @@ metadata:
   name: minio-s3-auth
   namespace: '{{ .Values.namespace }}'
   annotations:
-    argocd.argoproj.io/sync-wave: "0"
+    argocd.argoproj.io/sync-wave: "2"
   labels:
     apimanager.apps.3scale.net/watched-by: apimanager
 type: Opaque
-data:
-  AWS_ACCESS_KEY_ID: '{{ .Values.AsMap | dig "minio" "accessKey" "minio" | b64enc }}'
-  AWS_SECRET_ACCESS_KEY: '{{ .Values.AsMap | dig "minio" "secretKey" "minio123"| b64enc }}'
-  AWS_BUCKET: '{{ .Values.AsMap | dig "minio" "bucket" "3scale-maas" | b64enc }}'
-  AWS_REGION: '{{ "us-east-1" | b64enc }}'
-  AWS_HOSTNAME: '{{ .Values.AsMap | dig "minio" "host" "minio.minio.svc:9000" | b64enc }}'
-  AWS_PROTOCOL: '{{ "HTTP" | b64enc }}'
-  AWS_PATH_STYLE: '{{ "true" | b64enc }}'
+stringData:
+  AWS_ACCESS_KEY_ID: minio
+  AWS_SECRET_ACCESS_KEY: minio123
+  AWS_BUCKET: 3scale-maas
+  AWS_REGION: us-east-1
+  AWS_HOSTNAME: minio.minio.svc:9000
+  AWS_PROTOCOL: HTTP
+  AWS_PATH_STYLE: 'true'
 {{- end }}

--- a/examples/models-as-a-service-3scale/3scale/values.yaml
+++ b/examples/models-as-a-service-3scale/3scale/values.yaml
@@ -21,6 +21,7 @@ apimanager:
     name: llm-metrics
     version: 0.1
 
+# TODO: Comment this out when the 3scale Health Checks are merged into `ai-accelerator`
 apimanagerChecker:
   name: "apimanager-checker"
   syncwave: "3"

--- a/examples/models-as-a-service-3scale/README.md
+++ b/examples/models-as-a-service-3scale/README.md
@@ -20,13 +20,9 @@ This example automates the deployment and configuration of the following compone
 
 ## Storage Requirements
 
-**3scale** supports various storage backends including S3. For the specific requirement of RWX persistent volumes, options include Network File System (NFS) or **OpenShift Data Foundation (ODF)**.
+**3scale** supports various storage backends including S3. For the specific requirement of `RWX` persistent volumes, options include Network File System (NFS) or **OpenShift Data Foundation (ODF)**.
 
-This example is configured to use **OpenShift Data Foundation (ODF)** as the default provider for the `RWX` storage class.
-
-If **ODF** is not available, the example allows the user to provide a pre-configured storage class on the cluster capable of provisioning `RWX` volumes. To use a preconfigured storage class, please provide the storage class name in the `threeScale.storageClassName` parameter in the [bootstrap chart](examples/models-as-a-service-3scale/helm-charts/maas-bootstrap/values.yaml) before proceeding.
-
-If you decide to use S3-compatible storage, this example can deploy a simple **MinIO** instance to be used with **3scale**. To enable the **MinIO** deployment, set the parameter `threeScale.storageClassName` to blank (`""`).
+This example is configured to use **MinIO** as a S3-compatible storage provider for 3scale. If **ODF** is available, the user can provide a pre-configured storage class on the cluster capable of provisioning `RWX` volumes. To use a preconfigured storage class, please provide the storage class name in the `threeScale.storageClassName` parameter in the [bootstrap chart](examples/models-as-a-service-3scale/helm-charts/maas-bootstrap/values.yaml) before proceeding.
 
 ## Prerequisites
 
@@ -192,6 +188,16 @@ oc logs -n model-serving deployment/llama-32-1b-instruct-cpu
 # Check developer user status
 oc get developeruser -n 3scale
 ```
+
+### Triggering the update of the 3scale Developer CMS portal
+
+```bash
+curl -vkL -X POST "https://upload-cms-3scale.<cluster-domain>/" \
+  -H "Content-Type: application/json" \
+  -d '{ "tenant": "maas" }'
+```
+
+This will start the pipeline `upload-cms` and refresh the developer portal with the latest version available on the git repository.
 
 ## Cleanup
 

--- a/examples/models-as-a-service-3scale/create-threescale-registry-auth-secret.sh
+++ b/examples/models-as-a-service-3scale/create-threescale-registry-auth-secret.sh
@@ -1,0 +1,40 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+namespace="3scale"
+
+if ! oc get namespace "$namespace" &>/dev/null; then
+    echo "Creating namespace $namespace..."
+    oc apply -f - <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: "$namespace"
+  labels:
+    argocd.argoproj.io/managed-by: openshift-gitops
+EOF
+fi
+
+# Extract existing cluster pull secret
+PULL_SECRET=$(oc extract secret/pull-secret -n openshift-config --keys=.dockerconfigjson --to=- | base64 -w 0)
+
+# Create secret using HEREDOC
+oc apply -f - <<EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: threescale-registry-auth
+  namespace: ${namespace}
+  annotations:
+    argocd.argoproj.io/sync-options: "Prune=false"
+    argocd.argoproj.io/compare-options: "IgnoreExtraneous"
+  labels:
+    rhoai-example: maas
+    rhoai-example-component: ${namespace}
+type: kubernetes.io/dockerconfigjson
+data:
+  .dockerconfigjson: ${PULL_SECRET}
+EOF
+
+echo "Secret created successfully"

--- a/examples/models-as-a-service-3scale/helm-charts/maas-bootstrap/templates/application-minio.yaml
+++ b/examples/models-as-a-service-3scale/helm-charts/maas-bootstrap/templates/application-minio.yaml
@@ -5,8 +5,6 @@ kind: Application
 metadata:
   name: minio
   namespace: {{ .Values.namespace | default .Release.Namespace }}
-  annotations:
-    argocd.argoproj.io/sync-options: Validate=false  
   labels:
       component: bootstrap
 spec:
@@ -18,4 +16,6 @@ spec:
   destination:
     server: https://kubernetes.default.svc
     namespace: minio
+  syncPolicy:
+    automated: {}
 {{- end }}

--- a/examples/models-as-a-service-3scale/helm-charts/maas-bootstrap/values.yaml
+++ b/examples/models-as-a-service-3scale/helm-charts/maas-bootstrap/values.yaml
@@ -13,7 +13,8 @@ threeScale:
   path: examples/models-as-a-service-3scale/3scale
   channel: threescale-2.15
   startingCSV: 3scale-operator.v0.12.4
-  storageClassName: "ocs-storagecluster-cephfs"
+#  storageClassName: "ocs-storagecluster-cephfs"
+  storageClassName: ""
 
 cmsUpload:
   path: examples/models-as-a-service-3scale/3scale-config/cms-upload

--- a/examples/models-as-a-service-3scale/minio-instance/create-bucket-job.yaml
+++ b/examples/models-as-a-service-3scale/minio-instance/create-bucket-job.yaml
@@ -3,9 +3,9 @@ kind: Job
 metadata:
   name: minio-create-bucket
   annotations:
-    "helm.sh/hook": "post-install,post-upgrade"
-    "helm.sh/hook-delete-policy": "before-hook-creation,hook-succeeded"
-    "helm.sh/hook-weight": "6"
+    argocd.argoproj.io/hook: PostSync
+    argocd.argoproj.io/hook-delete-policy: HookSucceeded
+    argocd.argoproj.io/sync-wave: "10"
 spec:
   completions: 1
   template:
@@ -13,8 +13,24 @@ spec:
       name: minio-create-bucket
     spec:
       restartPolicy: Never
+      initContainers:
+      - name: wait-for-minio-service
+        image: wait4x/wait4x
+        command:
+          - "wait4x"
+          - "tcp"
+          - "minio.minio.svc:9000"
+          - "--timeout"
+          - "60s"
+        resources:
+          requests:
+            cpu: 100m
+            memory: 100Mi
+          limits:
+            cpu: 100m
+            memory: 100Mi
       containers:
-      - name: minio-bucket
+      - name: create-bucket
         image: quay.io/minio/mc
         env:
           - name: MC_CONFIG_DIR
@@ -24,12 +40,12 @@ spec:
           - name: MINIO_ACCESS_KEY
             valueFrom:
               secretKeyRef:
-                name: minio
+                name: minio-s3-auth
                 key: AWS_ACCESS_KEY_ID
           - name: MINIO_SECRET_KEY
             valueFrom:
               secretKeyRef:
-                name: minio
+                name: minio-s3-auth
                 key: AWS_SECRET_ACCESS_KEY
           - name: MINIO_BUCKET
             value: "3scale-maas"
@@ -43,3 +59,10 @@ spec:
               echo "Creating bucket" && \
               mc mb "minio-3scale/$MINIO_BUCKET" --ignore-existing && \
               echo "Bucket created"
+        resources:
+          requests:
+            cpu: 100m
+            memory: 100Mi
+          limits:
+            cpu: 100m
+            memory: 100Mi

--- a/examples/models-as-a-service-3scale/minio-instance/kustomization.yaml
+++ b/examples/models-as-a-service-3scale/minio-instance/kustomization.yaml
@@ -5,12 +5,14 @@ namespace: minio
 
 resources:
   - namespace.yaml
+  - secret.yaml
+  - create-bucket-job.yaml
 
 helmCharts:
   - name: minio
     releaseName: minio-release
     namespace: minio
-    version: 0.1.2
+    version: 0.1.3
     repo: https://redhat-ai-services.github.io/helm-charts/
     skipTests: true
     valuesFile: values.yaml

--- a/examples/models-as-a-service-3scale/minio-instance/secret.yaml
+++ b/examples/models-as-a-service-3scale/minio-instance/secret.yaml
@@ -1,0 +1,8 @@
+kind: Secret
+apiVersion: v1
+metadata:
+  name: minio-s3-auth
+type: Opaque
+stringData:
+  AWS_ACCESS_KEY_ID: minio
+  AWS_SECRET_ACCESS_KEY: minio123

--- a/examples/models-as-a-service-3scale/minio-instance/values.yaml
+++ b/examples/models-as-a-service-3scale/minio-instance/values.yaml
@@ -6,8 +6,7 @@ persistence:
   size: 50Gi
 
 credentials:
-  # existingSecret: my-minio-credentials
-  generateSecret: true
+  existingSecret: minio-s3-auth
   minioUserKey: AWS_ACCESS_KEY_ID
   minioPasswordKey: AWS_SECRET_ACCESS_KEY
 

--- a/examples/models-as-a-service-3scale/remove-example.sh
+++ b/examples/models-as-a-service-3scale/remove-example.sh
@@ -1,0 +1,81 @@
+#!/usr/bin/env bash
+
+# ApplicationSet
+echo "Deleting ApplicationSet..."
+oc delete appset models-as-a-service-3scale-helm  -n openshift-gitops --wait=false &> /dev/null
+oc delete appset models-as-a-service-3scale-kustomize -n openshift-gitops --wait=false &> /dev/null
+
+echo "Deleting Applications..."
+oc delete -n openshift-gitops apps 3scale cms-upload minio model-serving models-as-a-service-3scale-maas-bootstrap --wait=false &> /dev/null
+
+echo "Deleting AppProject..."
+oc delete appproject models-as-a-service-3scale -n openshift-gitops --wait=false &> /dev/null
+
+# 3scale
+echo "Deleting 3scale resources..."
+if oc get namespace 3scale &>/dev/null; then
+    # Define resources that require both patching and deleting
+    patch_and_delete_resources=(
+        applicationauth
+        application.capabilities.3scale.net
+        developeraccount
+        developeruser
+        activedoc
+        proxyconfigpromote
+        product
+        backend
+        apimanager
+        custompolicydefinition
+        subscription
+	    pipeline
+        pipelinerun
+        taskrun
+        tasks
+        eventlistener
+        pod
+        job
+        route
+        service
+        # rolebindings
+        # serviceaccount
+        # secret
+        # configmap
+    )
+
+    for item in "${patch_and_delete_resources[@]}"; do
+        echo "Removing finalizers from '$item' instances and deleting them..."
+        oc get "$item" -n 3scale -o name | xargs oc patch -p '{"metadata":{"finalizers":[]}}' -n 3scale --type=merge
+        oc get "$item" -n 3scale -o name | xargs oc delete -n 3scale
+    done
+
+    oc wait --for=delete apimanager/apimanager --timeout=60s
+    oc delete namespace 3scale || true
+fi
+
+# # Keycloak
+# echo "Deleting Keycloak resources..."
+# if oc get namespace redhat-sso &>/dev/null; then
+#     oc get keycloak -n redhat-sso -o name | xargs oc delete -n redhat-sso --wait=false
+#     oc get statefulset -n redhat-sso -o name | xargs oc delete -n redhat-sso --wait=false
+#     oc get subscription -n redhat-sso -o name | xargs oc delete -n redhat-sso
+#     oc delete namespace redhat-sso || true
+# fi
+
+# Minio
+echo "Deleting Minio resources..."
+if oc get namespace minio &>/dev/null; then
+    oc get deployment -n minio -o name | xargs oc delete -n minio --wait=false
+    oc delete namespace minio || true
+fi
+
+# LLM
+echo "Deleting LLM resources..."
+if oc get namespace llm-hosting &>/dev/null; then
+    oc get servingruntime -n llm-hosting -o name | xargs oc delete -n llm-hosting --wait=false
+    oc get inferenceservice -n llm-hosting -o name | xargs oc delete -n llm-hosting --wait=false
+    oc delete namespace llm-hosting || true
+fi
+
+# Helm release information
+echo "Deleting Helm release information..."
+oc get secret -n openshift-gitops -l name=models-as-a-service-3scale -l owner=helm -o name | xargs oc delete -n openshift-gitops || true

--- a/scripts/bootstrap.sh
+++ b/scripts/bootstrap.sh
@@ -17,7 +17,7 @@ choose_example(){
     echo "Choose an example you wish to deploy?"
     PS3="Please enter a number to select an example folder: "
 
-    select chosen_example in $(find "${examples_dir}" -mindepth 1 -maxdepth 1 -type d -exec basename {} \;); 
+    select chosen_example in $(find "${examples_dir}" -mindepth 1 -maxdepth 1 -type d -exec basename {} \; | sort);
     do
     test -n "${chosen_example}" && break;
     echo ">>> Invalid Selection";
@@ -40,20 +40,20 @@ choose_example_kustomize_option(){
 
     # Find the first directory matching the pattern ${chosen_example_path}/*/overlays
     overlays_parent_dir=$(find "${chosen_example_path}" -mindepth 2 -maxdepth 2 -type d -name "overlays" | head -n 1)
-    
+
     if [ -n "$overlays_parent_dir" ]; then
         overlays_dir="$overlays_parent_dir"
-        
+
         echo "Found overlays directory: ${overlays_dir}"
-        
+
         overlay_count=$(find "${overlays_dir}" -mindepth 1 -maxdepth 1 -type d | wc -l)
         if [ "$overlay_count" -gt 1 ]; then
             # multiple overlay options found
             # let the user choose which one to deploy
             echo "Multiple overlay options found in ${overlays_dir}:"
             PS3="Choose an option you wish to deploy?"
-            
-            select chosen_option in $(find "${overlays_dir}" -mindepth 1 -maxdepth 1 -type d -exec basename {} \;);
+
+            select chosen_option in $(find "${overlays_dir}" -mindepth 1 -maxdepth 1 -type d -exec basename {} \; | sort);
             do
                 test -n "${chosen_option}" && break;
                 echo ">>> Invalid Selection";
@@ -114,12 +114,12 @@ deploy_example(){
 
 set_repo_url(){
     GITHUB_URL=$(get_git_basename)
-    
+
     echo
     echo "Current repository URL: ${GITHUB_URL}"
     echo
     read -r -p "Press Enter to use this URL, or enter a new repository URL: " user_input
-    
+
     if [ -n "$user_input" ]; then
         GITHUB_URL="$user_input"
         echo "Updated repository URL to: ${GITHUB_URL}"
@@ -135,7 +135,7 @@ set_repo_branch(){
     echo "Current repository branch: ${GIT_BRANCH}"
     echo
     read -r -p "Press Enter to use this branch, or enter a new repository branch: " user_input
-    
+
     if [ -n "$user_input" ]; then
         GIT_BRANCH="$user_input"
         echo "Updated repository branch to: ${GIT_BRANCH}"

